### ISSUE TITLE
Add timestamps to log messages

### ISSUE THREAD

**Issue by nuest** [o2r-project/o2r-meta#47]
I've just tested metadata extraction as part of the upload process, which the test compendium "with metadata" takes over 30 seconds to complete. I _assume_ this is because of the online requests, since there is also a timeout for the first author (see log below), but it would be very useful if the **log statements would include an absolute timestamp** or a time difference for the last call (see example output below).

```
loader [aQNXk] Extracting metadata +0ms
  loader [aQNXk] Running metadata extraction with command "python3 ../o2r-meta/o2rmeta.py -debug extract --inputdir /tmp/o2r/compendium/aQNXk/data --outputdir /tmp/o2r/compendium/aQNXk/data/.erc --metafiles --ercid aQNXk --basedir /tmp/o2r/compendium/" +0ms
  loader [aQNXk] Completed metadata extraction:
  loader
  loader [o2rmeta][debug] received arguments: {'ercid': 'aQNXk', 'stayoffline': False, 'basedir': '/tmp/o2r/compendium/', 'debug': True, 'metafiles': True, 'modexml': False, 'outputdir': '/tmp/o2r/compendium/aQNXk/data/.erc', 'outputtostdout': False, 'inputdir': '/tmp/o2r/compendium/aQNXk/data', 'tool': 'extract'}
[o2rmeta] launching extractor
[o2rmeta][extract] directory </tmp/o2r/compendium/aQNXk/data/.erc> will be created during extraction...
[o2rmeta][extract] processing /tmp/o2r/compendium/aQNXk/data/erc.yml
[o2rmeta][extract] processing /tmp/o2r/compendium/aQNXk/data/document.Rmd
[o2rmeta][extract] requesting orcid for <Ted Tester>
[o2rmeta][extract] http orcid request: timeout
[o2rmeta][extract] requesting orcid for <Carl Connauthora>
[o2rmeta][extract] 200 OK
[o2rmeta][extract] 3504 bytes written to ../../../../../tmp/o2r/compendium/aQNXk/data/.erc/meta_2017091916535337_document_.json
[o2rmeta][extract] processing /tmp/o2r/compendium/aQNXk/data/Dockerfile
[o2rmeta][extract] processing /tmp/o2r/compendium/aQNXk/data/document.html
[o2rmeta][extract] 4 files processed
[o2rmeta][extract] requesting doi via crossref.org ...
[o2rmeta][extract] 200 OK
[o2rmeta][extract] debug: <get_doi_http> GET
[o2rmeta][extract] 4532 bytes written to ../../../../../tmp/o2r/compendium/aQNXk/data/.erc/metadata_raw.json
[o2rmeta][extract] downloading current erc spec
[o2rmeta][extract] saved </tmp/o2r/compendium/aQNXk/data/.erc/erc_spec.zip>

  loader  +47s
```

Depending on where the issue is, it might be useful to be able to configure a timeout (I'd rather not have an ORCID than wait 20 seconds for it, but if I can get it in 2 seconds that's fine) for external requests, not just turning them on or off.

**Comment by nuest** [comment]
I can confirm now that the issue was in the requests to the ORCID API, so the log statements could stay as they are, though I still find this a useful thing to have.

**Comment by nuest** [comment]
Just a small observation: The time of 47 seconds roughly fits the [used timeout](https://github.com/o2r-project/o2r-meta/blob/dev/extract/metaextract.py#L103) of 20 seconds per request + a few seconds for the other extraction stuff.

**Comment by 7048730** [comment]
included with 1f627ae894602218c99a2ec8859c5d0430e17b98
